src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs

```diff
@@ -1267,16 +1267,20 @@ public static EventLogHandle ConvertToSafeHandle(UnsafeNativeMethods.EvtVariant
                 return new EventLogHandle(val.Handle, true);
         }
 
-        public static Array ConvertToArray<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]T>(UnsafeNativeMethods.EvtVariant val, int size) where T : struct
+        public static Array ConvertToArray<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]T>(
+            UnsafeNativeMethods.EvtVariant val,
+            int size,
+            Func<int, T[]> createArray)
+            where T : struct
         {
             IntPtr ptr = val.Reference;
             if (ptr == IntPtr.Zero)
             {
-                return Array.CreateInstance(typeof(T), 0);
+                return createArray(0);
```

Comment:
```suggestion
                return new T[0];
```

Reply:
Good call. I'm not sure what I was thinking originally.